packages/vscode-mdx/src/typescript-plugin.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @typedef {import('typescript/lib/tsserverlibrary.js').server.Project} Project
+ * @typedef {import('typescript/lib/tsserverlibrary.js').server.PluginModuleFactory} PluginModuleFactory
+ * @typedef {import('typescript/lib/tsserverlibrary.js').LanguageService} LanguageService
+ */
+
+import {
+  decorateLanguageService,
+  decorateLanguageServiceHost,
+  searchExternalFiles
+} from '@volar/typescript'
+import {createFileProvider, resolveCommonLanguageId} from '@volar/language-core'
+import {getLanguageModule} from '@mdx-js/language-service'
+import {loadPlugins} from '../../language-server/lib/configuration.js'
```

Comment:
It might be nicer to move this function into `@mdx-js/language-service`. Although it’s not ideal `loadPlugins` depends on Node.js APIs.